stageleft_macro/src/lib.rs

```diff
@@ -0,0 +1,365 @@
+use proc_macro2::{Punct, Spacing, Span, TokenStream};
+use quote::{quote, quote_spanned, ToTokens};
+use syn::punctuated::Punctuated;
+use syn::spanned::Spanned;
+use syn::{AngleBracketedGenericArguments, PathArguments, Token, Type};
+
+mod quote_impl;
+
+#[proc_macro]
```

Comment:
I think this needs some documentation in the ///. I ended up here while trying to write some hydroflow+ code, so I assume other people would too.